src/ds/d500/hw_monitor_extended_buffers.cpp

```diff
@@ -50,43 +34,62 @@ namespace librealsense
     std::vector<uint8_t> hw_monitor_extended_buffers::send(command const & cmd, hwmon_response* p_response, bool locked_transfer) const
     {
         hwm_buffer_type buffer_type = get_buffer_type(cmd);
-        if (buffer_type == hwm_buffer_type::standard)
+        switch( buffer_type)
+        {
+        case hwm_buffer_type::standard:
             return hw_monitor::send(cmd, p_response, locked_transfer);
-
-        if (buffer_type == hwm_buffer_type::big_buffer_to_receive)
-            return send_big_buffer_to_receive(cmd, p_response, locked_transfer);
-
-        // hwm_buffer_type::big_buffer_to_send is the last remaining option
-        send_big_buffer_to_send(cmd, p_response, locked_transfer);
+        case hwm_buffer_type::extended_receive:
+            return extended_receive(cmd, p_response, locked_transfer);
+        case  hwm_buffer_type::extended_send:
+            extended_send(cmd, p_response, locked_transfer);
+        default:
+            return std::vector<uint8_t>();
+        }
         return std::vector<uint8_t>();
     }
 
-    std::vector<uint8_t> hw_monitor_extended_buffers::send_big_buffer_to_receive(command cmd, hwmon_response* p_response, bool locked_transfer) const
+    std::vector<uint8_t> hw_monitor_extended_buffers::extended_receive(command cmd, hwmon_response* p_response, bool locked_transfer) const
     {
-        int recv_msg_length = get_msg_length(cmd);
-        uint16_t overall_chunks = get_number_of_chunks(recv_msg_length);
-
         std::vector< uint8_t > recv_msg;
-        for (int i = 0; i < overall_chunks; ++i)
+
+        // send first command with 0/0 on param4, as we don't know the table size
+        // actual size will be returned as part for the response header and will be used
+        // to calculate the extended loop range
+        auto ans = hw_monitor::send(cmd, p_response, locked_transfer);
+        recv_msg.insert(recv_msg.end(), ans.begin(), ans.end());
```

Comment:
if I understand correctly, this is an API change:
instead of getting the first chunk, we get the message, in which only the header is needed, and then, we get the chunks in other messages.
Please confirm this change has been implemented also in HKR side.

Reply:
As discusses,
Current FW : we get the last chunk size of the first messages which is wrong, but since we get a valid header we extract the size from there and restart the querying process.

FW should supply a fix where 0/0 we send the first chunk with full size, then we can optimize and take this chunk and continue the loop from index 1.

This will be done on another PR once we have their fix.